src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/WindowsNodeMangler.cs

```diff
@@ -68,9 +75,59 @@ public sealed override string TypeGenericDictionary(TypeDesc type)
             return GenericDictionaryNamePrefix + NameMangler.GetMangledTypeName(type);
         }
 
-        public override string MethodGenericDictionary(MethodDesc method)
+        public sealed override string MethodGenericDictionary(MethodDesc method)
         {
             return GenericDictionaryNamePrefix + NameMangler.GetMangledMethodName(method);
         }
+
+        public sealed override string ExternMethod(string unmangledName, MethodDesc method)
+        {
+            if (_target.Architecture != TargetArchitecture.X86)
+            {
+                return unmangledName;
+            }
+
+            UnmanagedCallingConventions callConv;
+            if (method.IsPInvoke)
+            {
+                callConv = method.GetPInvokeMethodCallingConventions() & UnmanagedCallingConventions.CallingConventionMask;
+            }
+            else if (method.IsUnmanagedCallersOnly)
+            {
+                if (method is not Internal.TypeSystem.Ecma.EcmaMethod)
+                    callConv = method.Signature.GetStandaloneMethodSignatureCallingConventions();
+                else
+                    callConv = method.GetUnmanagedCallersOnlyMethodCallingConventions() & UnmanagedCallingConventions.CallingConventionMask;
```

Comment:
This part is a little bit weird for non-ECMA methods. Calling convention should not be a part of `MethodDesc.Signature`. It is not part of it for anything that comes out of a file.

Looks like we started placing it there for non-ECMA methods in https://github.com/dotnet/runtimelab/commit/5223594e0e5a8bb87f359a4952b10cb47c63e55d so it works out, but it's weird. It was already weird before this PR, but we just added another place that now depends on it.

The real fix would be to make `GetUnmanagedCallersOnlyMethodCallingConventions` a virtual method that whatever non-ECMA method with `IsUnmanagedCallersOnly == true` can override, but I don't understand the original motivation for placing this into signature, so maybe it wouldn't work for those cases.